pageserver/src/layered_repository.rs

```diff
@@ -968,12 +1064,34 @@ impl Timeline for LayeredTimeline {
 }
 
 impl LayeredTimeline {
+    fn get_checkpoint_distance(&self) -> u64 {
+        let tenant_conf = self.tenant_conf.read().unwrap();
+        tenant_conf
+            .checkpoint_distance
+            .unwrap_or(self.conf.default_tenant_conf.checkpoint_distance)
+    }
+
+    fn get_compaction_target_size(&self) -> u64 {
+        let tenant_conf = self.tenant_conf.read().unwrap();
+        tenant_conf
+            .compaction_target_size
+            .unwrap_or(self.conf.default_tenant_conf.compaction_target_size)
```

Comment:
> .unwrap_or

I see that `TenantConfOpt` is used in the layered repository only, and then its values are either used (if present), or defaults are used instead.
That all feels a bit odd, cannot we merge those values once during the init? 

We sure still need the `TenantConfOpt` to parse things from requests/configs, but why do we store it, not the `TenantConf`?

Reply:
I agree that it looks strange, but this implementation allows us to preserve information about what tenant-specific settings are set. `persist_tenant_conf()` only saves overridden parameters for the same reason.
This is necessary allow global config updates.
I.e. we want to update `gc_horizon` in PageServerConf for all tenants that do not use custom values.
Not sure, if this scenario will be used very often, because it requires a pageserver restart, but technically it is possible.